### PR TITLE
Integrated test coverage and codecov closes #1760

### LINKED ISSUE #1760: Integrate Coveralls for test coverage tracking into our CI
**Description:**

https://coveralls.io/

### PR DESCRIPTION
**Pull request by mkungla**
Test coverage is localy browsable and found in coverage/report
Added travis after_success: for posting report to codecov.

Added devDependencies,
istanbul, browserify-istanbul, codecov, karma-coverage

Updated devDependencies,
browserify: ^11.0.1 -> ^13.1.0
karma: ^0.13.15 -> 1.2.0
karma-browserify: ^4.4.0 -> ^5.1.0
karma-firefox-launcher: ^0.1.6 -> ^1.0.0
karma-mocha: ^0.2.0 -> ^1.1.1
karma-mocha-reporter: ^1.1.0 -> ^2.1.0
karma-sinon-chai: ^1.1.0 -> ^1.2.3
lolex: ^1.4.0 -> ^1.5.1
mocha: ^2.3.3 -> ^3.0.2
sinon: ^1.17.3 -> ^1.17.5

I wasn't familiar with [Coveralls](https://coveralls.io/), so used [codecov.io](https://codecov.io/) instead

### CODE REVIEW

**Comment by mkungla** [comment]
**You can test this PR out,**
*Create new branch with username prefix since it is easier to remember what branch is what*
`git checkout -b mkungla-coverage master`
*Pull and merge my coverage branch*
`git pull https://github.com/mkungla/aframe.git coverage`

now if you run tests new directory called `coverage/report` is created which is static html so you can see what your test are covering.

File: tests/karma.conf.js [hunk 1]
```diff
@@ -25,11 +36,26 @@ module.exports = function (config) {
...
+    frameworks: ['mocha', 'sinon-chai', 'chai-shallow-deep-equal', 'browserify'],
     preprocessors: {
-      'tests/**/*.js': ['browserify', 'env']
+      'tests/**/*.js': ['browserify', 'env'],
+      'src/**/*.js': ['coverage']
     },
-    reporters: ['mocha']
+    reporters: ['mocha', 'coverage'],
+    coverageReporter: {
+      dir: 'coverage',
```
Comment (ngokevin) [inline tests/karma.conf.js]: Can we put this in `tests/coverage/`?

**Comment by ngokevin** [comment]
Works great!!

File: tests/karma.conf.js (hunk 1 above)
Comment (mkungla) [inline tests/karma.conf.js]: yes

**Pull request merged**